### DISCUSSION TITLE
2007/04/30/%E4%BC%A0%E5%80%BC%E5%92%8C%E4%BC%A0%E5%BC%95%E7%94%A8

### DISCUSSION THREAD

**Discussion by giscus** [ruanqizhen/blog#156] [Announcements]
# 2007/04/30/%E4%BC%A0%E5%80%BC%E5%92%8C%E4%BC%A0%E5%BC%95%E7%94%A8

在现在常用的文本编程语言（C++, Java, C#）中，调用子函数时的传参方式主要是传引用方式，就是说，告诉被调用的函数的是参数所在的位置，而不是参数的数据。C++ 为了保持和 C 语言的兼容，一般的简单数据还是使用值传递，但对于大块的数据，比如数组，字符串，结构，类等等，也基本上都是引用形式传递的。

https://blog.qizhen.xyz/2007/04/30/%E4%BC%A0%E5%80%BC%E5%92%8C%E4%BC%A0%E5%BC%95%E7%94%A8

**Comment by ruanqizhen** [comment]
之前博客下的留言：


Qizhen
2007年05月11日 08:08 编辑
简单数据类型的长度是固定的，就不用说了。对于数组、字符串这样的数据类型。LabVIEW 对它们实际保存数据的那块内存的管理，和STL中的容器相类似，会自动根据需要调整占用内存的大小。就好像C语言，把一个Vector容器的引用传给子函数，即当输入又做输出是一样的。对于可以利用来做缓存重用内存，当然是再利用，不必释放。对于不再会用到的内存才会释放。

回复
杰
2007年05月10日 20:15 编辑
“在某些时候，一个节点得到了输入数据，LabVIEW 如果能够确认这个输入数据的内存肯定不会被其他部分的程序代码使用到，并且恰好节点的一个输出需要一块内存，LabVIEW 就不在为输出数据令开辟一块内存了，而是使用那个输入数据所在的内存。这叫做缓存重用。”，这个没太理解。如果输入数据所占内存小于这个输出要使用的内存，是不是就重开了？如果大于，那么是不是浪费了内存？而且，困惑的是，LV开辟了内存使用了之后，不释放吗？
抽空给俺解惑一下？
“如果一定要在 LabVIEW 代码中实现传引用，可以通过以下的方法：做一个全局数组变量，把数据存在数组里。VI 间传递的信息是数据在数组中的索引，一个表示序号整数值，就相当于是这块数据的引用。这样，所有对块数据的操作都是在同一内存中的，并且不同线程可以同时对这块内存做修改。 ”
这样的话，我感觉还是有危险。能保证一写多读的情况下还是可以用，如果多写多读，很容易就乱了。呵呵。
 

回复
arrowpig
2007年05月1日 08:44 编辑
给大虾挑一个刺 ：“LabVIEW 如果能够确认这个输入数据的内存肯定不会被其他部分的程序代码<BOLD>使用到</BOLD>”中的<BOLD>使用到</BOLD>应该改成<BOLD>所修改</BOLD>. -:)

回复
